flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveBatchSchedulerFactory.java

```diff
@@ -216,6 +224,13 @@ public SchedulerNG createInstance(
         }
     }
 
+    public static InputConsumableDecider.Factory loadInputConsumableDeciderFactory(
+            boolean onlyConsumeFinishedPartition) {
+        return onlyConsumeFinishedPartition
+                ? AllFinishedInputConsumableDecider.Factory.INSTANCE
+                : DefaultInputConsumableDecider.Factory.INSTANCE;
```

Comment:
This is hard to understand, because the semantic of `onlyConsumeFinishedPartition` and `AllFinishedInputConsumableDecider` are not equivalent. I understand that this is an intermediate state, but we'd better to explain it in comment.